BackEndLib/Wchar.cpp

```diff
@@ -130,6 +131,13 @@ void AsciiToUnicode(const std::string &str, WSTRING &wstr) {
 	AsciiToUnicode(str.c_str(), wstr);
 }
 
+//*****************************************************************************
+const WSTRING AsciiToUnicode(const char *psz) {
```

Comment:
Returning by `const` value doesn't really help here, assuming `WSTRING` is the same as [this one](https://github.com/CaravelGames/drod/blob/master/BackEndLib/Wchar.h#L77).

```suggestion
WSTRING AsciiToUnicode(const char *psz) {
```

I would also look into `[[nodiscard]]` for function signatures like these. It is a C++17 feature but there are equivalent compiler-specific attributes since C++98. It produces a warning when the return value of `AsciiToUnicode` is discarded, avoidining mistakenly assuming that `psz` gets mutated rather than a new value being returned:

```suggestion
[[nodiscard]] WSTRING AsciiToUnicode(const char *psz) {
```

Reply:
The `[[nodiscard]]` suggestion applies to pretty much every function here, or every function returning a value that should never be ignored.